internal/ingress/controller/kong.go

```diff
@@ -415,6 +412,12 @@ func (n *KongController) toDeckContent(
 
 	return &content
 }
+
+func sortByString(slice interface{}, fieldFn func(i int) string) {
```

Comment:
> refactor(controller): implement sortByString
> This REVERSES the sorting order from reverse to normal.
> The history behind reverse order is purely accidental. There are no
> known reasons in favor of one or the other. What matters is the order
> being deterministic.

Could you help me to better understand the impetus for this change, just for my own edification? If there's no known reasons in favor of one versus the other, is this change intended simply to make sure our ordering is intentional going forward for consistency? Or is there further context at play here?

Reply:
`toDeckContent()` generates a big struct that gets eventually fed into `Kong`. #404 added an optimization: after that PR, KIC refrains from feeding a new config to Kong if that config's hash hasn't changed since last known version.

In order to ensure that equivalent configs hash in a predictable manner, it's important to guarantee that the order of elements in slices (otherwise irrelevant) remains deterministic. Hence sorting.

The order of sorted elements doesn't matter though (as long as it's deterministic). #404 happened to implement sorting in reverse lexicographic order. I found the "reverse" bit unnecessarily confusing. Therefore,

> is this change intended simply to make sure our ordering is intentional going forward for consistency

is an accurate depiction of the situation.